python-api/tests/integrationtests2/test_controller.py

```diff
@@ -0,0 +1,337 @@
+"""
+Integration Tests for the dbus Controller
+"""
+
+from __future__ import absolute_import, print_function, unicode_literals
+import os, sys, datetime, time, random
+from .baseclass import IntegrationTestbase
+from mock import Mock
+
+class TestEstablishConnection(IntegrationTestbase):
+    """ Test starting and connecting to the Server
+    """
+
+    def test_establish_connection(self):
+        """ Start the Server and connect to it. Assert it starts up correctly
+        and can be connected to via the python-controller.
+        """
+        # setup_server asserts internally that the server started up correctly
+        self.setup_server()
+
+        # setup_controller asserts internally that the connection is okay
+        self.setup_controller()
```

Comment:
Generally this is frowned upon because if someone removes the asserts from those methods this test will continue to pass when it shouldn't.